libc/src/__support/UInt.h

```diff
@@ -103,13 +103,20 @@ template <size_t Bits, bool Signed> struct BigInt {
       val[i] = words[i];
   }
 
-  template <typename T, typename = cpp::enable_if_t<cpp::is_integral_v<T> &&
-                                                    sizeof(T) <= 16 &&
-                                                    !cpp::is_same_v<T, bool>>>
-  LIBC_INLINE constexpr explicit operator T() const {
-    if constexpr (sizeof(T) <= 8)
-      return static_cast<T>(val[0]);
+  template <typename T> LIBC_INLINE constexpr explicit operator T() const {
```

Comment:
does this template need any sort of limit (e.g. `enable_if_t<is_integral_v<T>>`)?

Reply:
We don't need extra guard here, since it is already guarded with `to<T>()` templates.

Reply:
yeah, if it doesn't fail here it will fail there.  I don't feel strongly either way, so merging for now.  If you feel strongly otherwise I'll send another patch to add that.